script/update-json-schema.sh

```diff
@@ -1,5 +1,26 @@
 #!/bin/bash
 
-NEW_HASH=$(git ls-remote git://github.com/department-of-veterans-affairs/vets-json-schema/ refs/heads/master | awk '{print $1}')
-sed -i '' -e 's/\(vets-json-schema\.git#\)\(.*\)\(",$\)/\1'$NEW_HASH'\3/' package.json
+# Find the versions
+NEW_VERSION=$(git ls-remote --tags git://github.com/department-of-veterans-affairs/vets-json-schema/ | awk -F / '{ print $3 }' | sort -V -r | head -n 1)
```

Comment:
I feel like this might require a bit of an explanation:
1) `git ls-remote --tags ...`: Fetch all the tags
2) `awk -F / ...`: Set the separator to `/` and print the third item--always the tag since the format is always `hash refs/tags/tagName`
3) `sort -V -r`: Sort by version in reverse order, so the latest is at the top
4) `head -n 1`: Take only the first entry (the latest version since we sorted in reverse order)

Only thing I'm not sure of there is if BSD `sort` has `-V`. Can somebody verify for me?

Reply:
BSD `sort` has the `-V` flag:

```
-V, --version-sort
             Sort version numbers.  The input lines are treated as file names in form PREFIX VERSION SUFFIX, where SUFFIX matches the regular expression "(.([A-Za-z~][A-Za-
             z0-9~]*)?)*".  The files are compared by their prefixes and versions (leading zeros are ignored in version numbers, see example below).  If an input string does not match
             the pattern, then it is compared using the byte compare function.  All string comparisons are performed in C locale, the locale environment setting is ignored.
```

Reply:
Excellent! Thanks. :smile: